lib/tom_queue/worker_supervisor.rb

```diff
@@ -0,0 +1,81 @@
+module TomQueue
+  class WorkerSupervisor
+
+    attr_accessor :processes, :before_fork, :after_fork
+
+    def initialize
+      @processes = {}
+      @before_fork = -> { }
+      @after_fork = -> { }
+      # TODO setup_signal_handling
```

Comment:
Not sure when to setup signal handling (in the #run method or the initializer?) What extra signal handling do we need?

Reply:
I'd be inclined to suggest during the `run` method, as this is where control of the main thread is handed over to the supervisor.

We just need to ensure we proxy the signals through to the child process, ensure it shuts-down cleanly!